src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from "react";
+import { Person } from "../types";
+import { getPeople } from "../api";
+import { Loader } from "./Loader";
+import { Link, useParams } from "react-router-dom";
+import classNames from "classnames";
+
+
+export const PeoplePage = () => {
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const { personSlug } = useParams();
+
+  const personInfo = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const personName = (name: string | null) => {
+    return people.find(person => person.name === name);
+  }
+
+
+  useEffect(() => {
+   setLoading(true)
+   getPeople()
+   .then(data => {
+    setPeople(data)
+    setLoading(false)
+   })
+   .catch(err => {
+     setError('Failed to load people data')
+     setLoading(false)
+   })
+  }, [])
+
+
+  if (loading) {
+    return <Loader />;
+  }
+
+  if (error) {
+    return (
+      <p data-cy="peopleLoadingError" className="has-text-danger">
+        Something went wrong
+      </p>
+    );
+  }
+
+  if (!people.length) {
+    return <p data-cy="noPeopleMessage">There are no people on the server</p>;
+  }
+
+
+  return (
+    <div>
+      <h1 className="title">People Page</h1>
+      <table
+        data-cy="peopleTable"
+        className="table is-striped is-hoverable is-narrow is-fullwidth"
+      >
+        <thead>
+          <tr>
+            {personInfo.map(info => (
+              <th key={info}>{info}</th>
+            ))}
+          </tr>
+        </thead>
+
+        <tbody>
+          {people.map((person) => {
+            const findMother = personName(person.motherName);
+            const findFather = personName(person.fatherName);
```

Comment:
You are calling 'personName' function for each person in the 'people' array to find the mother and father. This will result in a time complexity of O(n^2) which can be a performance issue if the 'people' array is large. It would be better to create a map with person's name (or a unique identifier like 'slug' or 'id') as key and the person object as value before rendering the 'people' array. Then you can use this map to find the mother and father in constant time.